pkg/entrypoint/options.go

```diff
@@ -0,0 +1,115 @@
+/*
+Copyright 2018 The Knative Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package entrypoint
+
+import (
+	"encoding/json"
+	"errors"
+	"flag"
+	"time"
+
+	"github.com/knative/build/pkg/entrypoint/wrapper"
+)
+
+// NewOptions returns an empty Options with no nil fields
+func NewOptions() *Options {
+	return &Options{
+		Options: &wrapper.Options{},
+	}
+}
+
+// Options exposes the configuration necessary
+// for defining the process being watched and
+// where in the image repository an upload will land.
+type Options struct {
+	// Args is the process and args to run
+	Args []string `json:"args"`
+	// Timeout determines how long to wait before the
+	// entrypoint sends SIGINT to the process
+	Timeout time.Duration `json:"timeout"`
+	// GracePeriod determines how long to wait after
+	// sending SIGINT before the entrypoint sends
+	// SIGKILL.
+	GracePeriod time.Duration `json:"grace_period"`
+	// ArtifactDir is a directory where test processes can dump artifacts
+	// for upload to persistent storage (courtesy of sidecar).
+	// If specified, it is created by entrypoint before starting the test process.
+	// May be ignored if not using sidecar.
+	ArtifactDir string `json:"artifact_dir,omitempty"`
+
+	*wrapper.Options
+}
```

Comment:
this seems super similar to https://github.com/kubernetes/test-infra/blob/master/prow/entrypoint/options.go - we should at least give a shout-out to it probably.

also are we 100% sure we need all the same options? (e.g. we don't do automatic artifact uploading afaik, do we need `ArtifactDir`?)